src/products/service/stock.service.ts

```diff
@@ -0,0 +1,33 @@
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
+import { StockRepository } from "../repository/stock.repository";
+import { OrderItem } from "src/orders/dto/create-order.dto";
+import { Stock } from "../domain/stock.entity";
+
+export interface StockItem{
```

Comment:
**issue:** keep types separated from the service classes

Reply:
Noted.
I actually removed those interfaces as I was not actually using them in the current version of the code.